core/src/environment.rs

```diff
@@ -151,6 +151,26 @@ impl<K: Hash + Eq, V: PartialEq> Environment<K, V> {
     fn was_cloned(&self) -> bool {
         Rc::strong_count(&self.current) > 1
     }
+
+    /// Checks quickly if two environments are obviously equal (when their components are
+    /// physically equal as pointers or obviously equal such as being both empty).
+    pub(crate) fn ptr_eq(this: &Self, that: &Self) -> bool {
+        // Check if the pointer to the previous layers are physically equal or both `None`
+        let prev_layers_eq = match (&*this.previous.borrow(), &*that.previous.borrow()) {
+            (Some(ptr_this), Some(ptr_that)) => {
+                // Check that the current layers are equal, either both equal as pointers or both empty
```

Comment:
previous layers, right? Actually, given the preceding comment I think this one is just unnecessary.

Reply:
Oops, indeed, this whole comment should actually move to line 168. Or be deleted.